src/logic/calculate.js

```diff
@@ -0,0 +1,91 @@
+// Calculate.js
+import { useState } from 'react';
+
+export function useCalculatorLogic() {
+  const [inputValue, setInputValue] = useState('0');
+  const [previousValue, setPreviousValue] = useState(null);
+  const [operator, setOperator] = useState(null);
+
+  const handleNumberClick = (number) => {
+    if (inputValue === '0' || inputValue === 'Error') {
+      setInputValue(number.toString());
+    } else {
+      setInputValue(inputValue + number);
+    }
+  };
+
+  const handleOperatorClick = (op) => {
+    if (op === '%') {
+      setOperator(op);
+      setPreviousValue(inputValue);
+      setInputValue('0');
+    } else {
+      setOperator(op);
+      setPreviousValue(inputValue);
+      setInputValue('0');
```

Comment:
- Good job done so far, but according to the project requirements, please use the [operate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js) file and [calculate.js](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/calculate.js) file provided in the curriculum. Please fix this 🙏